cpp/src/arrow/csv/writer.cc

```diff
@@ -59,6 +59,18 @@ namespace csv {
 
 namespace {
 
+// Copy delimiter or end-of-line chars. Though configurable,
+// delimiter is always one char, end-of-line is often at most two chars.
```

Comment:
Explain that this is for performance, stating the reason doing things this way is faster?